auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -64,7 +74,17 @@
   private Map<String, List<String>> lastMetadata;
 
   public GoogleAuthLibraryCallCredentials(Credentials creds) {
-    this.creds = checkNotNull(creds, "creds");
+    checkNotNull(creds, "creds");
+
+    // Avoid instanceof since that requires the class to exist. Only after we know there's an
+    // instance of the class is it reasonably safe to assume the class exists.
+    for (Class<?> clazz = creds.getClass(); clazz != null; clazz = clazz.getSuperclass()) {
```

Comment:
If you try loading the ServiceAccountCredentials class at initialization time of GoogleAuthLibraryCallCredentials, then you can avoid this for-loop by using Class.isAssignableFrom.